platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -122,6 +129,9 @@ class ConfigurationCacheProblems(
             if (seenSerializationErrorOnStore) {
                 return true
             }
+            if (areDegradationReasonsPresent()) {
```

Comment:
❓ I guess this is here to honor degradation requests that happen during serialization?

Reply:
Not really. Initial motivation is that `shouldDiscardEntry` flag is being checked even after serialization(or its absence), so this condition allow to return `true` when degradation was requested. Sorry for literally saying what the code tells, but maybe I'm not understood the question.